01-JS-DOM-APIs/Ex03/script.js

```diff
@@ -0,0 +1,32 @@
+function reqConfig (method, url, sync) {
+  this.method = method;
+  this.url = url;
+  this.sync = sync;
+}
+function newQuery () {
+  return new reqConfig ("GET", "http://api.icndb.com/jokes/random", true);
+}
+function reRequest(object) {
+  let req;
+  return new Promise (function(resolve, reject) {
+    req = new XMLHttpRequest();
+    req.open(object.method, object.url, object.sync);
+    req.send();
+    req.onload = function () {
+      if (this.status === 200 && this.readyState === 4) {
```

Comment:
Not need to check the readyState, once onload is executed the readyState is equals to 4.